sgt-back-end/index.js

```diff
@@ -0,0 +1,150 @@
+const express = require('express');
+const app = express();
+app.use(express.json());
+
+const pg = require('pg');
+
+// only create ONE pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  SELECT *
+  FROM grades
+  `;
+  db.query(sql)
+    .then(result => {
+      return res.json(result.rows);
+    })
+    .catch(err => {
+      if (err) {
+        // console.log(err);
+        res.status(500).json({
+          error: 'An unexpected error occurred.'
+        });
+      }
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  // return res.json(req.body);
+  // Line 37 is "object destructuring"
+  // const {
+  //   name,
+  //   course,
+  //   score
+  // } = req.body;
+  const inputName = req.body.name;
+  const inputCourse = req.body.course;
+  const inputScore = Number(req.body.score);
+  if (inputName === '' || inputCourse === '' || inputScore === '') {
+    return res.status(400).json({
+      error: 'Invalid/Missing name, course, or score.'
+    });
+  }
+  if ((!inputScore > 0 && !inputScore < 100)) {
+    return res.status(400).json({
+      error: 'Score must be an integer from 0 to 100.'
+    });
+  }
+  const sql = `
+    INSERT INTO "grades" ("name", "score", "course")
+    VALUES ($inputName, $inputScore, $inputCourse)
+    `;
+  // console.log(sql);
+  const params = [inputName, inputScore, inputCourse];
```

Comment:
For the placeholders in the SQL string you use actual numbers `$1` `$2` `$3` and so on.  `$1` will be replaced by the first thing in the `params` array, `$2` will be replaces by the second thing in the `params` array, `$3` will be replaced by the third thing in the `params` array and so on. Your code should look like:

```JAVASCRIPT
const sql = `
  INSERT INTO "grades" ("name", "score", "course")
  VALUES ($1, $2, $3)
`;
// console.log(sql);
const params = [inputName, inputScore, inputCourse];
```